cls/SourceControl/Git/Settings.cls

```diff
@@ -75,6 +81,36 @@ Method %OnNew() As %Status
     quit $$$OK
 }
 
+Method SaveWithSourceControl() As %Status
+{
+    set sc = $$$OK
+    #dim %SourceControl As %Studio.SourceControl.Interface
+    if '$isobject($get(%SourceControl)) {
+        do ##class(%Studio.SourceControl.Interface).SourceControlCreate($username)
+    }
+    set gitDir = ##class(%File).NormalizeDirectory(..namespaceTemp)_".git"
+    set skipSourceControl = ('##class(%File).DirectoryExists(gitDir) && 
+        (##class(%Studio.SourceControl.Interface).SourceControlClassGet() = ##class(SourceControl.Git.Extension).%ClassName(1)))
```

Comment:
This is a bit tricky because we could use a composite extension like in IPM or CodeTidy - that is, we could be using Embedded Git with the source control class set to one other than SourceControl.Git.Extension.